src/libraries/System.Security.Cryptography/tests/HashAlgorithmTestDriver.cs

```diff
@@ -320,6 +374,85 @@ protected void VerifyRepeating(string input, int repeatCount, string output)
             {
                 VerifyICryptoTransformStream(stream, output);
             }
+
+            using (Stream stream = new DataRepeatingStream(input, repeatCount))
+            {
+                VerifyOneShotStream(stream, output);
+            }
+
+            using (Stream stream = new DataRepeatingStream(input, repeatCount))
+            {
+                VerifyOneShotAllocatingStream(stream, output);
+            }
+        }
+
+        protected async Task VerifyRepeatingAsync(string input, int repeatCount, string output)
+        {
+            using (Stream stream = new DataRepeatingStream(input, repeatCount))
+            {
+                await VerifyComputeHashStreamAsync(stream, output);
+            }
+
+            using (Stream stream = new DataRepeatingStream(input, repeatCount))
+            {
+                await VerifyOneShotStreamAsync(stream, output);
+            }
+
+            using (Stream stream = new DataRepeatingStream(input, repeatCount))
+            {
+                await VerifyOneShotAllocatingStreamAsync(stream, output);
+            }
+        }
+
+        [Fact]
+        public void HashData_Null_Stream_Throws()
+        {
+            AssertExtensions.Throws<ArgumentNullException>("source", () => HashData((Stream)null));
+            AssertExtensions.Throws<ArgumentNullException>("source", () => HashData((Stream)null, Span<byte>.Empty));
+        }
+
+        [Fact]
+        public void HashData_ShortDestination_Throws()
+        {
+            AssertExtensions.Throws<ArgumentException>("destination", () => HashData(Stream.Null, Span<byte>.Empty));
+        }
+
+        [Fact]
+        public async Task HashDataAsync_Null_Stream_Throws()
+        {
+            await AssertExtensions.ThrowsAsync<ArgumentNullException>(
+                "source",
+                async () => await HashDataAsync((Stream)null, cancellationToken: default));
```

Comment:
The argument validation ones can/should be synchronous.

```suggestion
            Assert.Throws<ArgumentNullException>(
                "source",
                HashDataAsync((Stream)null, cancellationToken: default));
```